net/server/http_server_unittest.cc

```diff
@@ -1128,6 +1133,7 @@ TEST_F(CloseOnConnectHttpServerTest, ServerImmediatelyClosesConnection) {
   // closed without reading from it.
   EXPECT_FALSE(HasRequest());
```

Comment:
Did EXPECT_FALSE(HasRequest()) fail? Then it means our socket did send read request to the server, which sounds a little strange. Do you mean tcp_socket_starboard.cc calss SbSocketIsConnected() to determine if a connection was successful? In that case,  doesn't immediate close result in SbSocketIsConnected return false?

Reply:
This comment is copied directly from trunk, so I didn't really dig into it.

But it fails at line 234 ``EXPECT_THAT(client->ConnectAndWait(server_address_), IsOk());``, which triggers a crash from [tcp_socket_starboard.cc(532)] Check failed: SbSocketIsValid(socket_).